movebucket_test.go

```diff
@@ -216,6 +216,151 @@ func TestTx_MoveBucket(t *testing.T) {
 	}
 }
 
+func TestBucket_MoveBucket_DiffDB(t *testing.T) {
+	srcBucketPath := []string{"sb1", "sb2"}
+	dstBucketPath := []string{"db1", "db2"}
+	bucketToMove := "bucketToMove"
+	expectedErr := errors.ErrDifferentDB
+
+	var srcBucket *bbolt.Bucket
+
+	t.Log("Creating source bucket and populate some data")
+	db1 := btesting.MustCreateDBWithOption(t, &bbolt.Options{PageSize: pageSize})
+	err := db1.Update(func(tx *bbolt.Tx) error {
+		srcBucket = prepareBuckets(t, tx, srcBucketPath...)
+		return nil
+	})
+	db1.MustClose()
+	require.NoError(t, err)
+
+	t.Log("Creating target bucket and populate some data")
+	db2 := btesting.MustCreateDBWithOption(t, &bbolt.Options{PageSize: pageSize})
+	err = db2.Update(func(tx *bbolt.Tx) error {
+		prepareBuckets(t, tx, dstBucketPath...)
+		return nil
+	})
+	db2.MustClose()
+	require.NoError(t, err)
+
+	t.Log("Re-opening source bucket from DB1")
+	srcDB, sErr := bbolt.Open(db1.Path(), 0600, bbolt.DefaultOptions)
+	require.NoError(t, sErr)
+	defer func() {
+		cErr := srcDB.Close()
+		require.NoError(t, cErr)
+	}()
+
+	t.Log("Reading source bucket in a separate RWTx")
+	sTx, sErr := srcDB.Begin(true)
+	require.NoError(t, sErr)
+	defer func() {
```

Comment:
@ahrtr  the latest failure was due to `linter` 

the errors from `tx.Rollback()` must be checked, i wrapped them in anonymous func() to check the error and pass the linter